options_test.go

```diff
@@ -25,6 +25,108 @@ import (
 	models "github.com/firecracker-microvm/firecracker-go-sdk/client/models"
 )
 
+func TestGetFirecrackerConfig(t *testing.T) {
+	tempFile, err := ioutil.TempFile("", "firectl-test-drive-path")
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.Remove(tempFile.Name())
+	defer tempFile.Close()
+	cases := []struct {
+		opts        *options
+		expectedErr func(error) (bool, error)
+		outConfig   firecracker.Config
+	}{
+		// invalid metadata
+		{
+			opts: &options{
+				FcMetadata: "{ invalid:json",
+			},
+			expectedErr: func(e error) (bool, error) {
+				return e == invalidMetadata, invalidMetadata
+			},
+			outConfig: firecracker.Config{},
+		},
+		// error with getnetwork
+		{
+			opts: &options{
+				FcNicConfig: "no-slash",
+			},
+			expectedErr: func(e error) (bool, error) {
+				return e == parseNicConfigError, parseNicConfigError
+			},
+			outConfig: firecracker.Config{},
+		},
+		// error with getblock devices
+		{
+			opts: &options{
+				FcNicConfig:        "a/b",
+				FcAdditionalDrives: []string{"/no-suffix"},
+			},
+			expectedErr: func(e error) (bool, error) {
+				return e == invalidDriveSpecificationNoSuffix, invalidDriveSpecificationNoSuffix
+			},
+			outConfig: firecracker.Config{},
+		},
+		// error with parsevsocks
+		{
+			opts: &options{
+				FcNicConfig:        "a/b",
+				FcAdditionalDrives: []string{tempFile.Name() + ":ro"},
+				FcVsockDevices:     []string{"noCID"},
+			},
+			expectedErr: func(e error) (bool, error) {
+				return e == unableToParseVsockDevices, unableToParseVsockDevices
+			},
+			outConfig: firecracker.Config{},
+		},
+		// error with handle fifos
+		{
+			opts: &options{
+				FcNicConfig:        "a/b",
+				FcAdditionalDrives: []string{tempFile.Name() + ":ro"},
+				FcVsockDevices:     []string{"a:3"},
+				FcFifoLogFile:      "/etc/we-dont-have-access",
+			},
+			expectedErr: func(e error) (bool, error) {
+				return e == unableToCreateFifoLogFile, unableToCreateFifoLogFile
+			},
+			outConfig: firecracker.Config{},
+		},
+		// success
+		{
+			opts: &options{},
+			expectedErr: func(e error) (bool, error) {
+				return e == nil, nil
+			},
+			outConfig: firecracker.Config{
+				SocketPath: DefaultSocketPath,
+				Drives: []models.Drive{
+					models.Drive{
+						DriveID:      firecracker.String("1"),
+						PathOnHost:   firecracker.String(""),
+						IsRootDevice: firecracker.Bool(true),
+						IsReadOnly:   firecracker.Bool(false),
+					},
+				},
+				MachineCfg: models.MachineConfiguration{
+					HtEnabled: true,
+				},
+			},
+		},
+	}
+	for _, c := range cases {
```

Comment:
Can you do this with subtests so that `go test` tells you explicitly what fails?

```go
for _, c := range cases {
	t.Run(c.name, func(t *testing.T) {
		cfg, err := c.opts.getFirecrackerConfig()
		// ...
	})
}
```